// @vuadu/ipa-to-vie/dist/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LETTER_MAPPING = exports.ENDING_VOWEL_MAPPING = exports.NULL_MAPPING = exports.COMBINE_GRAVE = exports.COMBINE_DOT = exports.COMBINE_ACUTE = void 0;
exports.COMBINE_ACUTE = "\u0301";
exports.COMBINE_DOT = "\u0323";
exports.COMBINE_GRAVE = "\u0300";
exports.NULL_MAPPING = "_";
exports.ENDING_VOWEL_MAPPING = {
    a: "a",
    aɪ: "ai",
    aʊ: "ao",
    e: "e",
    eɪ: "ây",
    i: "i",
    ib: "ip",
    ih: "i",
    ik: "ich",
    im: "im",
    in: "in",
    ip: "ip",
    it: "it",
    iŋ: "inh",
    iɛ: "ia",
    iɡ: "ic",
    j: "i",
    jaʊ: "au",
    jeɪ: "ây",
    ji: "i",
    joʊ: "au",
    ju: "iu",
    jæ: "ye",
    jɑ: "ya",
    jɔ: "yo",
    jə: "ơ",
    jɛ: "ie",
    jɪ: "y",
    jʊ: "iu",
    o: "o",
    oʊ: "âu",
    u: "u",
    ub: "up",
    uh: "ơ",
    uk: "uc",
    um: "um",
    un: "un",
    up: "up",
    ut: "ut",
    uŋ: "oong",
    uɛ: "ue",
    uɡ: "uc",
    æ: "e",
    æb: "ep",
    æk: "ech",
    æm: "em",
    æn: "en",
    æp: "ep",
    æt: "et",
    æŋ: "eng",
    æɡ: "ec",
    ɑ: "a",
    ɑb: "ap",
    ɑh: "a",
    ɑk: "ac",
    ɑm: "am",
    ɑn: "an",
    ɑp: "ap",
    ɑt: "at",
    ɑŋ: "ang",
    ɑɛ: "",
    ɑɡ: "ac",
    ɔ: "o",
    ɔb: "op",
    ɔh: "o",
    ɔk: "ooc",
    ɔm: "om",
    ɔn: "on",
    ɔp: "op",
    ɔt: "ot",
    ɔŋ: "oong",
    ɔɡ: "ooc",
    ɔɪ: "oi",
    ə: "ơ",
    əb: "ơp",
    əh: "ơ",
    eɪt: "êt",
    əj: "ơi",
    ək: "ơc",
    əm: "ơm",
    ən: "ơn",
    əp: "ơp",
    ət: "ơt",
    əŋ: "ơng",
    əɡ: "ơc",
    ɛ: "e",
    ɛb: "ep",
    ɛh: "ê",
    ɛk: "êch",
    ɛm: "em",
    ɛn: "en",
    ɛp: "ep",
    ɛt: "et",
    ɛŋ: "êng",
    ɛɡ: "ec",
    ɪ: "i",
    ɪb: "ip",
    ɪh: "i",
    ɪk: "ich",
    ɪm: "im",
    ɪn: "in",
    ɪp: "ip",
    ɪt: "it",
    ɪŋ: "inh",
    ɪɛ: "ie",
    ɪɡ: "ich",
    jəŋ: "giang",
    ʊ: "u",
    ʊb: "up",
    ʊk: "uc",
    ʊm: "um",
    ʊn: "un",
    ʊp: "up",
    ʊt: "ut",
    ʊŋ: "uung",
    ʊɡ: "uc",
    eɪn: "ên",
};
exports.LETTER_MAPPING = {
    " ": "",
    ",": "",
    "/": "",
    a: "a",
    b: "b",
    d: "đ",
    dʒ: "d",
    e: "e",
    f: "ph",
    h: "h",
    i: "i",
    j: "d",
    k: "k",
    kw: "qu",
    m: "m",
    n: "n",
    o: "â",
    p: "p",
    s: "x",
    t: "th",
    tɹ: "tr",
    tʃ: "ch",
    u: "u",
    v: "v",
    w: "w",
    z: "d",
    æ: "e",
    ð: "đ",
    ŋ: "ng",
    ɑ: "a",
    ɔ: "o",
    ə: "ơ",
    ɛ: "ê",
    ɝ: "ơr",
    ɡ: "g",
    ɪ: "i",
    ɫ: "l",
    ɹ: "r",
    ʃ: "s",
    ʊ: "u",
    ʒ: "d",
    ˈ: " ",
    ˌ: "",
    θ: "th",
};
